src/components/Tree/child.js

```diff
@@ -107,6 +123,10 @@ export default function Child(props) {
     );
 }
 
+export default function Child(props) {
+    return <Consumer>{values => <ChildComponent {...values} {...props} />}</Consumer>;
+}
```

Comment:
it is more clean if we instead use the useContext hook, also since this component has only tree levels and children are no external component, we should simply pass props through and then avoid using context
https://reactjs.org/docs/context.html#before-you-use-context